### PR TITLE
query sensors at actual rate if we're logging

### LINKED ISSUE #954: Sensors logging at background sample rate even if we're logging
Looks like we're stuck at 50Hz for sensor channels even if we're logging data and sensor channels are > 50Hz. Need to bump up sample rate to actual logging rate if we're actually logging.

### PR DESCRIPTION
**Pull request by brentpicasso**
…ng; otherwise do the background rate. resolves #954

### CODE REVIEW

**Comment by RodADavison** [comment]
Would this be closer to the intended function?:
`if ((is_logging && currentTicks % loggingSampleRate == 0) || (!is_logging && currentTicks % BACKGROUND_SAMPLE_RATE == 0))`
rather than:
`if ((is_logging && currentTicks % loggingSampleRate == 0) || (currentTicks % BACKGROUND_SAMPLE_RATE == 0))`

File: src/logger/loggerTaskEx.c [hunk 1]
```diff
@@ -208,10 +208,11 @@ void loggerTaskEx(void *params)
                 /* Only reset the watchdog when we are configured and ready to rock */
                 watchdog_reset();
 
-                if (currentTicks % BACKGROUND_SAMPLE_RATE == 0)
+                const bool is_logging = logging_is_active();
+
+                if ((is_logging && currentTicks % loggingSampleRate == 0) || (currentTicks % BACKGROUND_SAMPLE_RATE == 0))
```
Comment (stieg) [inline src/logger/loggerTaskEx.c]: Wrap b/c line width.

**Comment by stieg** [comment]
@RodADavison Your code assumes that the logging rate is always a multiple of the background sample rate.  What if the background sample rate is faster than the logging sample rate?  Or what if they are not a multiple of one another.  Then it breaks down.

**Comment by RodADavison** [comment]
My suggestion was to fix the problems stieg mentions. I noticed the same issues.

File: src/logger/loggerTaskEx.c (hunk 1 above)
Comment (brentpicasso) [reply src/logger/loggerTaskEx.c]: ok.

**Comment by brentpicasso** [comment]
This logic line controls whether we refresh the ADCs and IMU for their current values - it doesn't actually create a sample record. 

@stieg  I think you're suggesting putting it below this code:

                /* Check if we need to actually populate the buffer. */
                const int sampledRate = populate_sample_buffer(sample,
                                                               currentTicks);
                if (sampledRate == SAMPLE_DISABLED)
                        continue;

If we do that, then we create the sample with old data from the ADC and IMU; having it above prepares the system with fresh sensor values. 

Also, this code change only augments the existing logic to ensure we refresh the ADCs and IMU at the logging rate or the background sample rate, ensuring we at least refresh the ADCs/IMU at the background sample rate.

**Comment by brentpicasso** [comment]
Timeout on PR. merging; low impact change.

**Pull request merged**